lib/io/can.cpp

```diff
@@ -1,8 +1,73 @@
 #include "can.hpp"
 
+#include <stdio.h>
+#include <unistd.h>
+
+#include <linux/can.h>
+#include <net/if.h>
+#include <sys/ioctl.h>
+#include <sys/socket.h>
+
 namespace hyped::io {
-Can::Can()
+Can::Can(hyped::core::ILogger &logger) : logger_(logger)
+{
+  if ((socket_ = socket(PF_CAN, SOCK_RAW, CAN_RAW)) < 0) {
+    logger_.log(hyped::core::LogLevel::kFatal, "Unable to open CAN socket");
+    return;
+  }
+
+  sockaddr_can socket_address;
+  socket_address.can_family  = AF_CAN;
+  socket_address.can_ifindex = if_nametoindex("can1");
+
+  if (socket_address.can_ifindex == 0) {
+    logger_.log(hyped::core::LogLevel::kFatal, "Unable to find CAN1 network interface");
+    close(socket_);
+    socket_ = -1;
+    return;
+  }
+
+  if (bind(socket_, (sockaddr *)&socket_address, sizeof(socket_address)) < 0) {
+    logger_.log(hyped::core::LogLevel::kFatal, "Unable to bind CAN socket");
+    close(socket_);
+    socket_ = -1;
+    return;
+  }
+
+  logger_.log(hyped::core::LogLevel::kInfo, "CAN socket successfully created");
+}
+
+int Can::sendCanFrame(can_frame message)
+{
+  if (socket_ < 0) {
+    logger_.log(hyped::core::LogLevel::kFatal,
+                "Trying to send CAN message but no CAN socket found");
+    return;
+  }
+
+  if (write(socket_, &message, sizeof(can_frame)) != sizeof(can_frame)) {
+    logger_.log(hyped::core::LogLevel::kFatal, "Failed to send CAN message");
+  }
+
+  logger_.log(hyped::core::LogLevel::kDebug,
+              "CAN message sent, ID:%i, DATA: %i %i %i %i %i %i %i %i",
+              (int)message.can_id,
```

Comment:
use `static_cast<int>` instead

Reply:
also this look ugly 

maybe we can add support for printing arrays to the logger? all we have to do is add a conversion from array to args (which is then converted to an array). Maybe a function `vlog` @kshxtij ?

Reply:
I'll look into it

Reply:
I think converting an array to va_args/va_lists doesn't seem like a good idea